src/wizard.ts

```diff
@@ -0,0 +1,83 @@
+import * as fs from 'fs'
+import { prompt, PromptObject } from 'prompts'
+
+interface ProjectOptions {
+  projectName: string
+  documentSyntax: string
+  entryPoint: string
+}
+
+export const getProjectNamePrompt = (
+  processDirectory: string,
+): PromptObject => ({
+  type: 'text',
+  name: 'projectName',
+  message: 'Project name',
+  initial: processDirectory,
+  validate: (input) =>
+    !!input.match(/^[\w\-_]+$/) ||
+    'Project name may only contain letters, hyphens and underscores',
+})
+
+export const getDocumentSyntaxPrompt = (): PromptObject => ({
+  name: 'documentSyntax',
+  type: 'select',
+  message: 'Document syntax',
+  choices: [
+    { title: 'json', value: 'json' },
+    { title: 'yaml', value: 'yaml' },
+  ],
+  initial: 0,
+})
+
+export const getEntryPointPrompt = (documentSyntax: string): PromptObject => ({
+  name: 'entryPoint',
+  type: 'text',
+  message: 'Entry point',
+  initial: `openapi.${documentSyntax}`,
+  validate: (input) =>
+    !!input.match(
+      new RegExp(
+        `(.*)(?<!\\.|${documentSyntax === 'json' ? 'yaml' : 'json'})$`,
```

Comment:
If you're using negative lookbehinds in regular expressions, make sure you include an `engines` field in the `package.json`, as older versions of node don't support this syntax.

Reply:
added >18.12.1, inline with the .nvmrc